operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/model/PollingLocationResponse.java

```diff
@@ -70,6 +98,30 @@ public void setError(List<Output> error) {
         this.error = error;
     }
 
+    public JsonObject getExtension() {
+        return extension;
+    }
+
+    public void setExtension(JsonObject extension) {
+        this.extension = extension;
+    }
+
+    public void addOperationOutcomeToExtension(OperationOutcome outcome) throws FHIRGeneratorException, IOException {
+        // Convert to the JsonObject and add as "operationOutcome" to the existing extension.
+        try (StringWriter writer = new StringWriter();) {
+            FHIRGenerator.generator(Format.JSON).generate(outcome, writer);
+
+            String outcomeString = writer.toString();
+            try (ByteArrayInputStream bais = new ByteArrayInputStream(outcomeString.getBytes());
+                    JsonReader jsonReader = JSON_READER_FACTORY.createReader(bais, StandardCharsets.UTF_8)) {
```

Comment:
we have `JsonSupport.toJsonObject(Resource)` for doing this part of it

Reply:
changed. thanks for pointing that out.